clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -73,6 +73,10 @@ public URI getUri() {
 
     @Override
     public void initialize(URI name, Configuration conf) throws IOException {
+        initialize(name, conf, new LakeFSClient(conf));
+    }
+
+    void initialize(URI name, Configuration conf, LakeFSClient lfsClient) throws IOException {
```

Comment:
Suggestion: different signature - mixing inheritance and overloading is never a good thing.

Reply:
I don't follow, sorry.  Are you suggesting changing the _name_ in addition to the signature?  Assuming "yes" and renaming this one.  Note that it is *not* public or protected, so I'm not sure overloading could ever break anything here.

Reply:
yes having two methods with the same name that one signature is set by inheritance and the other one (the protected) is not can be hard to read/debug  at runtime.